packages/synapse-react-client/src/components/TableColumnSchemaEditor/ColumnModelForm.tsx

```diff
@@ -176,340 +145,353 @@ export default function ColumnModelForm(props: ColumnModelFormProps) {
               disabled={disabled}
-              error={!!errorsByField['columnType']}
-            >
-              {allowedColumnTypes.map(value => {
-                return (
-                  <MenuItem value={value} key={value}>
-                    {getColumnTypeFriendlyName(value)}
-                  </MenuItem>
-                )
-              })}
-            </Select>
-            {errorsByField['columnType'] && (
-              <FormHelperText color={'error.main'}>
-                {errorsByField['columnType']}
-              </FormHelperText>
-            )}
-          </FormControl>
-        )}
-      </Box>
-      <Box my={isDefaultColumn ? 'auto' : undefined}>
-        {isDefaultColumn ? (
-          (columnModel as ColumnModelFormData).maximumSize ?? ''
-        ) : (
-          <TextField
-            value={(columnModel as ColumnModelFormData).maximumSize ?? ''}
-            disabled={disabled || !canHaveSize(columnModel.columnType)}
-            InputProps={{
-              disableInjectingGlobalStyles:
-                DISABLE_INJECTING_GLOBAL_STYLES_VALUE,
+              fullWidth
+              error={!!errorsByField['name']}
+              helperText={errorsByField['name']}
+            />
+          )}
+        </Box>
+        <Box my={isDefaultColumn ? 'auto' : undefined}>
+          {isDefaultColumn ? (
+            getColumnTypeFriendlyName(columnModel.columnType)
+          ) : (
+            <FormControl fullWidth>
+              <Select
+                label="Column Type"
+                value={columnModel.columnType}
+                onChange={e => {
+                  dispatch({
+                    type: 'changeColumnModelType',
+                    columnModelIndex,
+                    jsonSubColumnModelIndex: jsonSubColumnIndex,
+                    newColumnType: e.target.value as ColumnTypeEnum,
+                  })
+                }}
+                slotProps={{
+                  input: {
+                    'aria-label': 'Column Type',
+                  },
+                }}
+                sx={fieldSx}
+                disabled={disabled}
+                error={!!errorsByField['columnType']}
+              >
+                {allowedColumnTypes.map(value => {
+                  return (
+                    <MenuItem value={value} key={value}>
+                      {getColumnTypeFriendlyName(value)}
+                    </MenuItem>
+                  )
+                })}
+              </Select>
+              {errorsByField['columnType'] && (
+                <FormHelperText color={'error.main'}>
+                  {errorsByField['columnType']}
+                </FormHelperText>
+              )}
+            </FormControl>
+          )}
+        </Box>
+        <Box my={isDefaultColumn ? 'auto' : undefined}>
+          {isDefaultColumn ? (
+            (columnModel as ColumnModelFormData).maximumSize ?? ''
+          ) : (
+            <FieldWithRecommendedMinimum
```

Comment:
For columnModel.maximumSize, use a new component FieldWithRecommendedMinimum. Pass in a recommended value, and the original value if they exist.